### ISSUE TITLE
Sparse losses return nan when there is -inf in the input

### ISSUE THREAD

**Issue by erickrf** [deep-spin/entmax#9]
The sparse loss functions (and their equivalent classes) return nans when there is -inf in the input. 

Example:
```
import torch
import numpy as np
from entmax import entmax15_loss, sparsemax_loss
x = torch.rand(10, 5)
y = torch.randint(0, 4, [10])
x[:, 4] = -np.inf
entmax15_loss(x, y) 
# tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])

sparsemax_loss(x, y)
# tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])

**Comment by bpopeters** [comment]
https://github.com/deep-spin/entmax/blob/master/entmax/losses.py#L44

If anything in X is non-finite, it looks to me like this line will cause problems.

**Comment by vene** [comment]
good catch!

That line was a bit faster in my quick experiments when compared to other approaches but it should be avoidable.
